api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -219,6 +220,9 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Invalid solution name: catalog - %s", name), v1alpha2.BadRequest)
 					}
 
+					if label_key != "" && label_value != "" && annotation_name != "" {
```

Comment:
Why adding annotation depends on label env variables?

Reply:
labeling and annotation should be bundled

Reply:
From OSS perspective, we can treat them separately?